cards.py

```
@@ -0,0 +1,282 @@
+                }
+            )
+
+    if strict:
+        seen = set()
+        for action in normalized:
+            action_id = action["id"]
+            if action_id in seen:
+                raise DuplicateActionIdError(
+                    f"duplicate action id {action_id!r}: each action on a card "
+                    "must be addressable on its own"
+                )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)